File: main workshop/services/java/payment-processor/src/main/java/com/powertoolsride/paymentprocessor/model/PaymentResult.java
package com.powertoolsride.paymentprocessor.model;

import com.fasterxml.jackson.annotation.JsonProperty;

public record PaymentResult(
    @JsonProperty("success") boolean success,
    @JsonProperty("payment") Payment payment,
    @JsonProperty("transactionId") String transactionId,
    @JsonProperty("errorMessage") String errorMessage,
    @JsonProperty("processingTimeMs") long processingTimeMs
) {}

File: main-workshop/services/dotnet/payment-processor/Models/PaymentResult.cs
using System.Text.Json.Serialization;

public class PaymentResult
{
    [JsonPropertyName("success")]
    public bool Success { get; set; }
    [JsonPropertyName("payment")]
    public Payment Payment { get; set; }
    [JsonPropertyName("transactionId")]
    public string? TransactionId { get; set; }
    [JsonPropertyName("errorMessage")]
    public string? ErrorMessage { get; set; }
    [JsonPropertyName("processingTimeMs")]
    public long ProcessingTimeMs { get; set; }
    [JsonPropertyName("cachedResponse")]
    public bool CachedResponse { get; set; }

    [JsonPropertyName("cacheExpiration")]
    public long? CacheExpiration { get; set; }
}